socket_windows.go

```diff
@@ -0,0 +1,181 @@
+package kr
```

Comment:
Ok. I had some time tonight to look at this a little closer...   KrDir refactors really well into windows/unix variants.  However, it's not as simple for Listen and Dial.  I think we're going to need windows and unix variants of all 4 of the following:
- AgentListen
- DaemonListen
- HostAuthListen
- DaemonDial 

This seems like a cleaner abstract than needing to reference the <AGENT|DAEMONHOST_AUTH>_SOCKET_NAME constants elsewhere in he code.  

I've got the beginnings of a commit for this approach, but before I tidy it up and push it, I figured as what you think.  Thought?